enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,242 @@
+
+- OVS Hardware Offload will only be supported on bare metal deployment.
+
+- Multus supports overwriting cluster network definition by specifying a
+customized default net-attach-def in pod annotation. Customized net-attach-def
+will still use `ovn-k8s-cni-overlay` CNI plugin, but with an special annotation
+that requests a SR-IOV device.
+
+- RHCOS image shall be based on RHEL-8.3 for fully support of OVS Hardware
+Offload with connection track feature.
+
+- OpenvSwitch package version 2.13.
+
+- Mellanox SR-IOV VF-LAG provides hardware link-aggregation (LAG) for hardware
+offloaded ports. It allows offloading of OVS rules to a linux `bond` network
+device that combines two idential and hardware offload capable Physical
+Functions (PFs). In order to use SR-IOV VF-LAG, a bond interface needs to be
+created and used as geneve endpoint of OVS bridge. With the SR-IOV VF LAG
+capability, the NIC PFs can receive the rules that the OVS tries to offload to
+the linux bond net device and offload them to the hardware e-switch.
+
+#### RHCOS
+
+- RHEL 8.3 be used as RHCOS image base for fully support of OVS Hardware Offload
+with connection track feature.
+
+#### OpenvSwitch
+
+- OpenvSwitch package [version 2.13](https://github.com/openshift/ovn-kubernetes/pull/122) is currently used in OpenShift.
+
+#### OVN-Kubernetes
+
+- OVS Hardware Offload is a [feature available](https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/ovs_offload.md) in OVN-Kubernetes upstream.
+
+- `ovn-k8s-cni-overlay` CNI plugin configures the pod interface by moving the VF
+to pod namespace and plugging VF representor to `br-int` OVS bridge. The
+relevant [patches](https://github.com/openshift/ovn-kubernetes/commit/6c96467d0d3e58cab05641293d1c1b75e5914795) are available downstream.
+
+#### Cluster Network Operator
+
+- Cluster Network Operator will enable hardware offload and configure tc-policy
+on node OpenvSwitch instance and restart OpenvSwitch instance for hardware
+offload change to take effect.
+
+- Enabling `hw-offload` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:hw-offload=true`. The default value
+is `false`.
+
+- Setting `tc-policy` in CNO `ovs-node` DaemonSet is equivalent to executing
+`ovs-vsctl set Open_vSwitch . other_config:tc-policy=skip_sw`. The default
+value is `none`, optional values can be one of `none`, `skip_sw` and `skip_hw`.
+This field is only relevant if `hw-offload` is enabled.
+
+#### Multus
+
+- Multus will support specifying net-attach-def with customized annotations for
+[default cluster network in pod annotations](https://github.com/intel/multus-cni/blob/master/doc/configuration.md#specify-default-cluster-network-in-pod-annotations).
+Customized annotations will contain a resource name requesting SR-IOV device.
+
+#### SR-IOV Operator
+
+- SR-IOV Operator will support new device type `switchdev` in Operator
```

Comment:
Is there any udev rules created in order to retain the name of the PF device? After chaning the mode to `switchdev`, the PF devices name will be changed. Since the VF-Reps are already renamed, there is no need for udev rules for VF-reps, but PF naming has to be managed appropriately.

Reply:
Yes, PF will be renamed to its original name when provisioned to switchdev mode using udev rules. otherwise, bond interface (ifcfg-bond) will not be able to find the right interface.

Reply:
@moshe010 if using shared-gateway mode, do you think we still have this issue (NetworkManager restart interface which causes qdisc rules be removed)?

Reply:
I think so we solve it in openstack by adding udev rule [1] to unmanage the VF representors


[1] - https://github.com/openstack/os-net-config/blob/master/os_net_config/sriov_config.py#L355-L359

Reply:
I think so. In openstack we add udev rule to disable NM on VF represntors [1] 

[1] - https://github.com/openstack/os-net-config/blob/master/os_net_config/sriov_config.py#L355-L359

Reply:
Added a sentence in the config daemon section which will create/apply udev rules for both PF device and VF representors